infrastructure/src/main/java/org/corfudb/infrastructure/BatchProcessor.java

```diff
@@ -189,8 +192,42 @@ private void processor() {
                                 currOp.setResultValue(tails);
                                 break;
                             case LOG_ADDRESS_SPACE_QUERY:
-                                // Retrieve the address space for every stream in the log.
-                                currOp.setResultValue(streamLog.getStreamsAddressSpace());
+                                StreamsAddressRequest streamsAddressRequest =
+                                        (StreamsAddressRequest)currOp.getMsg().getPayload();
+                                StreamsAddressResponse streamsAddressResponse;
+
+                                // Currently, only servers for address space requests for sequencer bootstrap or
+                                // sequencer address space trim.
+                                switch (streamsAddressRequest.getReqType()) {
+                                    // servers for sequencer address space trim
+                                    case StreamsAddressRequest.STREAMS:
+                                        if (!streamsAddressRequest.isBeforeCompactionMark()) {
```

Comment:
Using compactionMark is not correct as we discussed, we need to use committed tail.

Reply:
Now I use the min of compactionMark and committedTail.